RadarSDK/RadarLocationPremissionsManager.m

```diff
@@ -0,0 +1,167 @@
+            RadarLocationPermissionsStatus *status = [[RadarLocationPermissionsStatus alloc] initWithStatus:self.locationManager.authorizationStatus
+                                                                                   backgroundPopupAvailable:NO
+                                                                                          inForegroundPopup:self.status.inForegroundPopup
+                                                                          userRejectedBackgroundPermissions: self.status.userRejectedBackgroundPermissions];
+            [self updateStatus:status];
+        }
+
+        // We set a flag that request has been made and start a timer. If we resign active we unset the timer.
+        // When the timer fires and the flag has not been unset, we assume that app never resigned active.
+        // Usually this means that the user has previously rejected the background permissions.
+        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.5 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
+            if (self->danglingBackgroundPermissionsRequest) {
+                if (@available(iOS 14.0, *)) {
+                    RadarLocationPermissionsStatus *status = [[RadarLocationPermissionsStatus alloc] initWithStatus:self.locationManager.authorizationStatus
+                                                                                           backgroundPopupAvailable:self.status.backgroundPopupAvailable
+                                                                                                  inForegroundPopup:self.status.inForegroundPopup
+                                                                                  userRejectedBackgroundPermissions:YES];
+                    [self updateStatus:status];
+                }
+            }
+            self->danglingBackgroundPermissionsRequest = NO;
+        });
+    }
+    if (!background && self.status.locationManagerStatus == kCLAuthorizationStatusNotDetermined) {
+        [self.locationManager requestWhenInUseAuthorization];
+        if (@available(iOS 14.0, *)) {
+            RadarLocationPermissionsStatus *status = [[RadarLocationPermissionsStatus alloc] initWithStatus:self.locationManager.authorizationStatus
+                                                                                   backgroundPopupAvailable:self.status.backgroundPopupAvailable
+                                                                                          inForegroundPopup:YES
+                                                                          userRejectedBackgroundPermissions: self.status.userRejectedBackgroundPermissions];
+            [self updateStatus:status];
+        }
+    }
+}
+
+- (void)applicationDidBecomeActive {
+    // we need to handle the case of double updates, we onyl want to update the status if and only if we are coming back from a popup and the status has changed.
+    if (inBackgroundLocationPopUp) {
```

Comment:
`if (self.inBackgroundLocationPopUp) {` ?

Reply:
`self` is implicit in instance methods. it will either be `inBackgroundLocationPopUp ` or `self -> inBackgroundLocationPopUp `